From 4f0bb978c14dd0a204a3cd3ded6aec385795f03b Mon Sep 17 00:00:00 2001
From: Dinesh Kumar Mirche <dmirche@habana.ai>
Date: Mon, 01 Jan 2024 16:18:02 +0200
Subject: [PATCH] [SW-0] uart: reduce ring buffer size by 2 bytes

This is a WA patch to pass the CI. As with ring buffer
size 512 the coral hangs hence CI failure. This can be
reverted once coral issue fixed.

Change-Id: Id7f7440681ed96d5db49d9a7720537122b363b9c
Signed-off-by: Dinesh Kumar Mirche <dmirche@habana.ai>
---

diff --git a/lib/common/uart.c b/lib/common/uart.c
index ec33350..b9c7ed5 100644
--- a/lib/common/uart.c
+++ b/lib/common/uart.c
@@ -16,7 +16,8 @@
 #define UART_SR_RXFULL			(1 << 1) /* RX Buffer Full */
 #define STR_UINT_SZ			8
 #define BITS_IN_SINGLE_CHAR		4
-#define RING_BUFF_SIZE			SZ_512
+/* TODO: revert this to SZ_512 once coral fixes the issue related to this */
+#define RING_BUFF_SIZE			(SZ_512 - SZ_2)
 
 struct device_uart uart = {
 	.regs_initialized = false,
